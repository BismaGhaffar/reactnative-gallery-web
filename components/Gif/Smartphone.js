import styled from 'styled-components'

const Smartphone = styled.div`
  position: relative;
  padding: 65px 12px;
  background: rgba(49, 49, 49, 0.8);
  border-radius: 38px;
  box-shadow: inset 0 0 3px 0 rgb(79, 86, 95, 0.2);
  border: solid 1px rgb(79, 86, 95, 0.2);
  max-width: 275px;
  min-width: ${({ minWidth }) => (minWidth && `${minWidth}px`) || '194px'};
  min-height: 250px;
  margin: auto;
  cursor: ${({ cursorPointer }) =>
    (cursorPointer ? 'pointer' : 'url(/static/images/play-circle.svg), auto;')};
  transform: ${({ rotate }) => (rotate ? 'rotate(-0.25turn)' : 'none')};
  margin-top: ${({ rotate }) => (rotate ? '-100px' : 'auto')};
`

export default Smartphone
