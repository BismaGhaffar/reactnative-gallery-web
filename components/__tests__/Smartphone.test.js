import React from 'react'
import renderer from 'react-test-renderer'
import Smartphone from '../Gif/Smartphone'

it('Smartphone can be created', () => {
  const comp = renderer.create(<Smartphone />)
  expect(comp).toBeDefined()
})

it('<Smartphone /> toMatchSnapshot', () => {
  const tree = renderer.create(<Smartphone />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('<Smartphone /> toMatchSnapshot 2', () => {
  const tree = renderer.create(<Smartphone cursorPointer />).toJSON()
  expect(tree).toMatchSnapshot()
})
