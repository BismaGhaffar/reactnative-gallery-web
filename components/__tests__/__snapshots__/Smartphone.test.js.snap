// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Smartphone /> toMatchSnapshot 1`] = `
<div
  className="Smartphone-s165niw0-0 dRCHgZ"
/>
`;

exports[`<Smartphone /> toMatchSnapshot 2 1`] = `
<div
  className="Smartphone-s165niw0-0 kMAOzs"
/>
`;
